### ISSUE TITLE
 The @Counted metric is always reusable

### ISSUE THREAD

**Issue by pfmackin** [oracle/helidon#1001]
Helidon Version: 1.2.0, 1.2.1, 1.2.2-SNAPSHOT  
Helidon MP 
JDK version: 1.8.0_211  
OS: MacOS 10.13.6  
Problem Description:  The @Counted metric is always reusable, even if I explicitly specify ‘reusable=false’ (which should be the default).

Steps to reproduce:
1) Create a quickstart project using MP archetype to test this.

2) Update GreetApplication.java and replace getClasses method:
		@Override
	public Set<Class<?>> getClasses() {
	return CollectionsHelper.setOf(GreetResource.class, GreetingCards.class);
	}

3) Add new class GreetingCards:

package io.helidon.examples.quickstart.mp;

import java.util.Collections;
import javax.enterprise.context.RequestScoped;
import javax.json.Json;
import javax.json.JsonBuilderFactory;
import javax.json.JsonObject;
import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;
import org.eclipse.microprofile.metrics.annotation.Counted;

@Path("/cards")
@RequestScoped
public class GreetingCards {

  private static final JsonBuilderFactory JSON = Json.createBuilderFactory(Collections.emptyMap());

  @GET
  @Path("/birthday")
  @Produces(MediaType.APPLICATION_JSON)
  @Counted(name = "specialEventCard", monotonic = true, reusable = false)
  public JsonObject birthdayCard() throws InterruptedException {
    return createResponse("Here are some birthday cards ...");
  }

  @GET
  @Path("/wedding")
  @Produces(MediaType.APPLICATION_JSON)
  @Counted(name = "specialEventCard", monotonic = true, reusable = false)
  public JsonObject weddingCard() throws InterruptedException {
    return createResponse("Here are some wedding cards ...");
  }

  private JsonObject createResponse(String msg) {
    return JSON.createObjectBuilder().add("message", msg).build();
  }
}

4) Build and run the application as follows:
mvn package -DskipTests=true
java -jar target/helidon-quickstart-mp.jar 

5) From a different terminal window, run these commands:

curl  http://localhost:8080/cards/birthday
…
curl  http://localhost:8080/cards/wedding
…
curl -H "Accept: application/json"  http://localhost:8080/metrics/application
{"io.helidon.examples.quickstart.mp.GreetingCards.specialEventCard":2}~>

**Comment by tjquinno** [comment]
Found several bugs:
- We were ignoring the `reusable` setting on annotations.
- We were not correctly initializing the reusability in the metadata for a new metric.
- We were incompletely checking for a mismatch between the reusability of a previously-registered metric and a new registration of the same name.

Fixes en route as part of the MP metrics 2.0 support.

**Comment by tjquinno** [comment]
Fixes for this are part of PR #992

**Comment by tjquinno** [comment]
I've gone through the reproducer steps. Now, when attempting to launch the server it displays this:
```
java.lang.IllegalArgumentException: Attempting to re-register metric io.helidon.examples.quickstart.mp.GreetingCards.specialEventCard that is already registered as non-reusable
```

**Comment by tjquinno** [comment]
I've reopened this issue because, although using the new metrics impl all is well, using the old impl the multiple registrations of the same counter are still (incorrectly) permitted.

**Comment by tjquinno** [comment]
Addressed by PR #1048.
